### PR TITLE
Add profile to usercontext

### PR DESCRIPTION
**Pull request by Brod8362**
This adds a field of `Option<ProfileData>` to the `UserContext` struct so manual calls to `get_profile` are no longer needed.

Additionally, some logic relating to profile address management has been incorporated as member functions of `ProfileData` to avoid code repetition. 

Addresses #60

### CODE REVIEW

File: src/backend/src/routes.rs [hunk 1]
```diff
@@ -289,55 +273,55 @@ pub struct PartialQuote {
...
-        //render quote history again
-        drop(submit_resp);
-        quote_history(pool, ctx).await
-    } else {
-        //fuel quote cant be submitted
-        Template::render(
-            "fuel_quote",
-            context!{
-                message: "Submission error"
+pub async fn quote_submit(ctx: &UserContext, pool: &State<Pool<Sqlite>>, form: Form<PartialQuote>) -> Result<Template, HtmxRedirect> {
```
Comment (Mnguyener) [inline src/backend/src/routes.rs:276]: Nice this looks a lot cleaner

File: src/backend/src/database.rs [hunk 2]
```diff
@@ -430,15 +425,22 @@ mod tests {
...
 
-        let quote_id = crate::database::store_quote(&pool, &user_id, 11.0, "myaddr", "2024-01-13", 0.0, 0.0).await.expect("fetch quote id");
+        let quote_id = crate::database::store_quote(&pool, &user_id, 11.0, &profile.one_line_address(), "2024-01-13", 0.0, 0.0).await.expect("fetch quote id");
         let quote_option = crate::database::get_quote(&pool, &quote_id).await.unwrap();
         assert!(&quote_option.is_some());
 
         let quote = quote_option.unwrap();
         assert!(&quote.gallons_requested == &11.0);
-        assert!(&quote.address == "address 1, address 2, city of leagues, florida 12345");
+        assert!(quote.address == String::from("address 1, address 2, city of leagues, florida 12345"));
```
Comment (Mnguyener) [inline src/backend/src/database.rs:443]: What's the purpose of making this line a `String`?

**Review by Mnguyener — DISMISSED** [review]
Looks good, just had a question

File: src/backend/src/database.rs (hunk 2 above)
Comment (Brod8362) [reply src/backend/src/database.rs:443]: Was a leftover from testing, has been restored

**Pull request merged**